core/sdk-core/src/main/java/software/amazon/awssdk/core/util/DefaultSdkAutoConstructList.java

```diff
@@ -158,4 +162,8 @@ public int lastIndexOf(Object o) {
         return impl.subList(fromIndex, toIndex);
     }
 
+    // For deserialization
+    private Object readResolve() {
```

Comment:
Can you give us more context of why this is needed?

Reply:
This is a serialization trick to ensure that we use the singleton instance (INSTANCE on line 37) when we deserialize this class's object. (Usually, deserialization constructs a new object without readResolve())

Reply:
@zoewangg ping

Reply:
Apologies for the delay. Hmm, I'm not sure if we want to add this. This seems a bit hacky to me.